### ISSUE TITLE
providedCompile configuration from war plugin is not added to testRuntimeClasspath

### ISSUE THREAD

**Issue by lfielke** [gradle/gradle#16406]
Gradle 7.0-milestone-2 seems to not add things from the `providedCompile` configuration to the `testRuntimeClasspath` when using the `war` plugin, which is different from the 6.8.3 behavior and contrary to how the documentation suggests it should work.

Examples below are using the Gradle project from this repo at https://github.com/gradle/gradle/tree/6477354f52b53b039f2d1bf019cd08406818bb0f/subprojects/docs/src/snippets/webApplication/customized/groovy

### Current Behavior
With Gradle 6.8.3, in a clone of this repo:

```
cd subprojects/docs/src/snippets/webApplication/customized/groovy
gradle dependencies --configuration testRuntimeClasspath
```

Shows:

```
testRuntimeClasspath - Runtime classpath of source set 'test'.
+--- :compile:1.0
|    +--- :compile-transitive-1.0
|    \--- :providedCompile-transitive:1.0
+--- javax.servlet:servlet-api:2.5
+--- :providedCompile:1.0
|    \--- :providedCompile-transitive:1.0
+--- :providedRuntime:1.0
+--- :runtime:1.0
\--- junit:junit:4.13
     \--- org.hamcrest:hamcrest-core:1.3
```

Note that dependency `:providedRuntime:1.0` is on that classpath.

### Steps to Reproduce 

Running the same command with `gradle-7.0-milestone-2`, shows:

```
testRuntimeClasspath - Runtime classpath of source set 'test'.
+--- :compile:1.0
|    +--- :compile-transitive-1.0
|    \--- :providedCompile-transitive:1.0
+--- :runtime:1.0
\--- junit:junit:4.13
     \--- org.hamcrest:hamcrest-core:1.3
```

Note that these dependencies aren't on the classpath any more:
```
+--- javax.servlet:servlet-api:2.5
+--- :providedCompile:1.0
|    \--- :providedCompile-transitive:1.0
+--- :providedRuntime:1.0
```
 
I'd expect the classpath to be the same as in gradle 6.

The impact of this change for us is that our tests no longer execute, because they depend on some `providedCompile` dependencies being on the runtime classpath. I think something related to this might be because the `compile` configuration from the `java` plugin appears to have been removed -- it's no longer listed in the docs at https://docs.gradle.org/7.0-milestone-2/userguide/java_plugin.html#sec:java_plugin_and_dependency_management

### Workaround

Manually make the `testRuntimeClasspath` extend `providedCompile`:

```groovy
configurations {
   testRuntimeClasspath.extendsFrom(providedCompile)
}
```

### Your Environment
Using gradle wrapper with `distributionUrl=https\://services.gradle.org/distributions/gradle-7.0-milestone-2-all.zip`
Build scan URL: https://scans.gradle.com/s/72yo2emruko7k

**Comment by donat** [comment]
Fixed in #16406.

**Comment by ljacomet** [comment]
Thanks for following up though it looks like it slipped through the cracks and 7.0 to 7.2 are impacted.
A fix is being worked on through #17415 to be released with Gradle 7.3.
